### DISCUSSION TITLE
Simulations just stopped without an explicit reason on the .out file

### DISCUSSION THREAD

**Discussion by IngSillo** [firemodels/fds#13683] [Q&A]
Dear all,

As the title of this thread suggests, I left my PC while testing two simulations. When I returned, the PC was still on, but there were no ongoing or completed cmdFDS processes.

I already checked the system logs for any accidental Windows Update restarts, but it appears the PC was neither shut down nor restarted.

I’m starting to think that my PC's RAM might not have been sufficient to run both simulations simultaneously, and perhaps the processes crashed as a result. I had only considered the CPU capacity, which I believed was adequate.

Shouldn’t such events be logged in the .out file? Mines end as shown below, which makes it seem as if the processes were shut down manually.

![image](https://github.com/user-attachments/assets/709b6421-6e47-4e94-8aab-9a161d729537)

![image](https://github.com/user-attachments/assets/05e40930-4ad0-40b9-ae5f-050b7c48fcef)

**Comment by marcosvanella** [comment]
Can you restart the runs and check the RAM usage? Also, I noted it is taking one and a half hours to run 100 time steps. This might be due to the OS having to use virtual memory in your hard drive (RAM full), which makes calculations much slower.

**Reply by marcosvanella** [reply]
Also, check the .err files to see if there is information on why the runs stopped.

**Comment by mcgratta** [comment]
FDS typically writes error message to "standard error". In old Fortran parlance, this is logical unit 0. When you run a job on Windows at the command prompt, this standard error "file" is dumped directly to the screen. We don't recommend that you run long jobs like this because these sessions are vulnerable to who-knows-what and errors are not permanently captured. You should redirect the output to a file. 

All that being said, if you post your input file, I can run it and estimate the RAM usage.

**Reply by IngSillo** [reply]
That is extremely interesting, how can someone redirect the output to a file? 

However I've re-run a single simulation. I checked that currently it employs 52% of the RAM of my PC (64 GB). Hence, I believe that there's no virtual space to run two simulations as I tried before.

Can you tell if lack of RAM memory is enough to kill the processes? Lastly, It would be interesting to read a review on which FDS namelists or inputs put the greatest load on RAM and CPU, respectively. This way, I could understand how to optimize my design.

**Reply by drjfloyd** [reply]
For typical FDS simulations where you have a single step fast reaction, RAM usage is driven mostly by the number of grid cells and if you have sprinklers the number of particles. Your out file snippet shows that your mesh is at least 597x363x47. That is at least 10 million cells and it is going to take a lot of RAM as you observed.

**Reply by IngSillo** [reply]
Indeed, according to task manager RAM is at 50%-53% during such simulation. 

Does this mean that if I expand the RAM of my workstation the computation will speed up?

Any reference to read on how to tackle and increase computational speed?

I've got a i9 raptor lake + 64 gb RAM.

I'm thinking to build up a Linux dedicated workstation only for FDS.

**Reply by drjfloyd** [reply]
More RAM will not increase speed. Faster RAM, if your current RAM is not as fast as your PC supports, might increase speed some. The i9 has multiple p-cores, so you could speed things up by splititing your single mesh into multiple meshes.

**Reply by IngSillo** [reply]
Yup, thanks for your reply. Unfortunately, I'm running a simulation with FDS+Evac which shouldn't support MPI Process according to the User Guide. That's why I'm searching for tricky solutions and references to speed up any other process of the simulation without using parallel processing.
